### ISSUE TITLE
Heal Selection not Working

### ISSUE THREAD

**Issue by rolodoom** [Diolinux/PhotoGIMP#49]
Ubuntu 20.04.2 LTS x86_64.
FLATPAK and GIMPO installed from Flathub

Heal Selection is not working. Here is the error text.

Traceback (most recent call last):
  File "/app/lib/gimp/2.0/python/gimpfu.py", line 740, in response
    dialog.res = run_script(params)
  File "/app/lib/gimp/2.0/python/gimpfu.py", line 361, in run_script
    return apply(function, params)
  File "/home/rolo/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/plugin-heal-selection.py", line 154, in heal_selection
    pdb.plug_in_resynthesizer(timg, tdrawable, 0,0, useBorder, work_drawable.ID, -1, -1, 0.0, 0.117, 16, 500)
error: procedure not found

**Comment by mwt** [comment]
I don't have this issue. Maybe it's fixed?

**Comment by cyberfunk** [comment]
Heal Selection does not even appear in the menu for me with Gimp 2.10.32 on macOS Monterey. If I install the resynthesizer plug-in separately replacing the version included with PhotoGIMP that version is broken and does not work although it does appear in the menu. Is there a specific version of Gimp and a specific version of macOS that PhotoGIMP is most compatible with?

**Comment by mwt** [comment]
> @cyberfunk 
> Is there a specific version of Gimp and a specific version of macOS that PhotoGIMP is most compatible with?

No idea. It's most compatible with Linux flatpak which this issue was about. It sounds like you have a completely different issue.

**Comment by cyberfunk** [comment]
> > @cyberfunk
> > Is there a specific version of Gimp and a specific version of macOS that PhotoGIMP is most compatible with?
> 
> No idea. It's most compatible with Linux flatpak which this issue was about. It sounds like you have a completely different issue.

Ooops...😅

**Comment by rolodoom** [comment]
> I don't have this issue. Maybe it's fixed?

I still have this issue using flatpak on Linux.  Same issue on Manjaro. I switched to Manjaro official gimp package with AUR plugins, and using Photogimp without the included plugins.

**Comment by cyberfunk** [comment]
My goodness even without resynthesizer GIMP has been a nightmare fo rme to use on Monterey. Certain features of the program simply do not work at all and overall the program is extremely buggy. For instance I can't filter fonts and I can't center elements. Those features simply don't work at all.🤦 No explanation, no troubleshooting, nothing. At first I thought maybe PhotoGIMP had somethign to do with it and I did a fresh install without any tweaks or customization. Same result. God bless all the professionals who can use GIMP as their daily driver.

**Comment by rolodoom** [comment]
I just find a solution. I deleted the plugin included with PhotoGIMP and Installed de resynthesizer plugin from flathub. It's working just fine on Manjaro.

**Comment by azuredusk10** [comment]
I also had this problem. Installing PhotoGIMP means that the Heal Selection filter doesn't work, it just throws a "procedure not found" error.

I didn't know how to uninstall the plugin, so I deleted GIMP using:
`flatpak remove --delete-data org.gimp.GIMP`

Then reinstalled GIMP from GNOME Software, using Flathub as the source.

Then installed "Resynthesizer" from GNOME Software (it's listed on the same page as GIMP, underneath the description).

Sadly this means I can't use PhotoGIMP as this is a key tool in my workflow.

**Comment by rolodoom** [comment]
Currently it is working for me on Manjaro, here it's what I did:

1. Delete `~/.config/GIMP/` folder
2.  Uninstall all GIMP software from Repositories and AUR
3. Install the flatpak version of GIMP and Resynthesizer from flathub:
`flatpak install flathub org.gimp.GIMP org.gimp.GIMP.Plugin.Resynthesizer `
4. Install PhotoGIMP as instructed on the folder `~/.var/`
5. Remove all  plugins from `~/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/`
**NOTE:** I deleted all Plugins 'cause I wont use this version. Only installing the ones from Flathub repository.

Tested and working on multiple machines running Manjaro Up to Date.
